File: python/lc14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = []
        for i in range(len(strs[0])):
            cur = strs[0][i]
            end = False
            for s in strs[1:]:
                if i >= len(s) or cur != s[i]:
                    end = True
                    break
            if end:
                break
            else:
                ans.append(cur)            

        return ''.join(ans)

File: cs/lc14.cs
public class Solution {
    public string LongestCommonPrefix(string[] strs) {
        var ans = new StringBuilder();
        for(int i=0;i<strs[0].Length;i++)
        {
            bool end = false;
            char prev = strs[0][i];
            for(int j = 1;j<strs.Length;j++)
            {
                if(i >= strs[j].Length || prev != strs[j][i])
                {
                    end = true;
                    break;
                }
            }
            if(end)
                break;
            else
                ans.Append(prev);
        }
        return ans.ToString();
    }
}
